build.gradle.kts

```
@@ -0,0 +1,232 @@
+if (project.version == "unspecified") {
+    version = "7.0.0-SNAPSHOT"
+}
+
+val website by extra("http://scoverage.org")
+val vcsUrl by extra("https://github.com/scoverage/gradle-scoverage.git")
+val scmUrl by extra("scm:git:$vcsUrl")
+val sonatypeUser by extra(System.getenv("SONATYPE_USER"))
+val sonatypePass by extra(System.getenv("SONATYPE_PASS"))
+
+gradlePlugin {
+    plugins {
+        create("gradleScoverage") {
+            id = "org.scoverage"
+            implementationClass = "org.scoverage.ScoveragePlugin"
+            displayName = "Gradle Scoverage plugin"
+        }
+    }
+}
+
+pluginBundle {
+    website = website
+    vcsUrl = vcsUrl
+    description = project.description
+    tags = listOf("coverage", "scala", "scoverage")
+}
+
+apply(plugin = "maven-publish")
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_1_8
+    targetCompatibility = JavaVersion.VERSION_1_8
+}
+
+dependencies {
+    compileOnly("org.scoverage:scalac-scoverage-plugin_2.13:1.4.2")
+    implementation(group = "commons-io", name = "commons-io", version = "2.6")
+
+    implementation(kotlin("script-runtime"))
+    testImplementation(kotlin("test"))
+
+    testImplementation("junit:junit:4.12")
+    testImplementation("org.junit.jupiter:junit-jupiter-api:5.5.2")
+    testRuntimeOnly("org.junit.vintage:junit-vintage-engine:5.5.2")
+
+    testImplementation("org.hamcrest:hamcrest:2.2")
+}
+
+sourceSets {
+    create("functionalTest") {
+        java.srcDir(file("src/functionalTest/java"))
+        compileClasspath += sourceSets["main"].output
+        runtimeClasspath += sourceSets["main"].output
+    }
+    create("crossScalaVersionTest") {
+        java.srcDir(file("src/crossScalaVersionTest/java"))
+        compileClasspath += sourceSets["main"].output + sourceSets["functionalTest"].output
+        runtimeClasspath += sourceSets["main"].output + sourceSets["functionalTest"].output
+    }
+}
+
+configurations {
+    named("functionalTestImplementation") {
+        extendsFrom(configurations["testImplementation"])
+    }
+    named("functionalTestRuntimeOnly") {
+        extendsFrom(configurations["testRuntimeOnly"])
+    }
+
+    named("crossScalaVersionTestImplementation") {
+        extendsFrom(configurations["testImplementation"])
+    }
+    named("crossScalaVersionTestRuntimeOnly") {
+        extendsFrom(configurations["testRuntimeOnly"])
+    }
+}
+
+val crossScalaVersionTest by tasks.registering(Test::class) {
+    description = "Runs the cross scala version functional test."
+    group = "verification"
+    testClassesDirs = sourceSets["crossScalaVersionTest"].output
+    classpath = sourceSets["crossScalaVersionTest"].runtimeClasspath
+    setForkEvery(1) // crucial to run every test in its own JVM
+
+    testLogging {
+        events("passed", "failed", "skipped")
+        showStandardStreams = System.getenv("CI") == "true"
+    }
+
+    mustRunAfter(tasks["test"])
```

```suggestion
    mustRunAfter(tasks.test)
```
Note that if a plugin is applied in the `plugins` block, its tasks created during the configuration time get typed accessors in the `.gradle.kts` files, which is the idiomatic way of accessing such tasks.